### DISCUSSION TITLE
Bandwidth Utilization

### DISCUSSION THREAD

**Discussion by jeffdfield** [microsoft/winget-cli#1133] [Ideas]
For large organizations I would be hesitant to use winget on a large scale. I'm hoping the package manager will be able to use at least one of the content sharing technologies already in place such as connected cache, branch cache, peer cache, delivery optimization or something else to share content with peers on the same network. We choose not to use chocolatey for this reason. I can't have 20,000 clients all downloading the same copy of Adobe Reader.

**Comment by denelon** [comment]
@jeffdfield, we have implemented support for Delivery Optimization (DO) in the client. We still have work to do on the service side. There are a few different changes that will take a bit longer to roll out due to the dependencies in Windows. The first is related to the client timeout behavior in DO. The current algorithm is a 30 minute exponential decay. We have asked for a "fast fail" mechanism for the Windows Package Manager. We currently fall back to "wininet" after a custom timeout. You can look at the settings schema to get an idea of the options.

The other is a bit more complex. The service uses HTTP range requests to support parallel download and Peer to Peer data transfer. I don't have all of the details yet, but I believe it also has other dependencies with at least one of the cache mechanisms. We are coordinating with the team to determine when they will be able to support our use case, and what version(s) of which Windows SKUs will support this.
